### ISSUE TITLE
Improve site responsiveness

### ISSUE THREAD

**Issue by kamicut** [developmentseed/scoreboard#110]
There's a delay when clicking on links, I suspect due to server side rendering taking a long time. We should improve the performance by [caching SSR'ed pages](https://github.com/zeit/next.js/tree/canary/examples/ssr-caching) and measuring performance.

**Comment by kamicut** [comment]
cc @scisco @sethvincent

**Comment by scisco** [comment]
yes, we should do that. We should probably add something visual feedback to the user on the frontend while one is waiting for a response.
